src/python_pachyderm/experimental/mixin/pfs.py

```diff
@@ -1254,6 +1257,71 @@ def path_exists(
 
         return True
 
+    def mount(self, mount_dir: str, repos: List[str] = []) -> None:
+        """Mounts Pachyderm repos locally.
+
+        Parameters
+        ----------
+        mount_dir : str
+            The directory to mount repos to.
+        repos : List[str], optional
+            The repos to mount. If empty, all repos are mounted.
+
+        Notes
+        -----
+        Mounting uses FUSE, which causes some known issues on macOS. For the
+        best experience, we recommend using mount on Linux. We do not support
+        mounting on macOS 1.11 and later.
+
+        Additionally, we recommend using mount in read-only access.
+
+        Examples
+        --------
+        >>> client.mount("dir_a", ["repo1", "repo2@staging"])
+        """
+        Path(mount_dir).mkdir(parents=True, exist_ok=True)
+
+        subprocess.run(
+            ["sudo", "pachctl", "unmount", mount_dir],
+            stdout=subprocess.DEVNULL,
+            stderr=subprocess.STDOUT,
+        )
+
+        cmd = ["pachctl", "mount", mount_dir]
+        for r in repos:
+            cmd.append("-r")
+            cmd.append(r)
+
+        subprocess.Popen(cmd, stdout=subprocess.DEVNULL, stderr=subprocess.STDOUT)
+        time.sleep(0.25)
```

Comment:
Any chance we could add a todo to replace the sleep with a health check of some kind (e.g. `ls` the mounted directory, and run ListFile, and compare the results to ensure equality)?

Reply:
Added a check. I find that even with the check, it's necessary to add a delay to give time for mount to complete.

Reply:
See comment below re:retries